pkgs/development/libraries/science/math/tiny-cuda-nn/default.nix

```diff
@@ -0,0 +1,156 @@
+{
+  stdenv,
+  cmake,
+  fetchFromGitHub,
+  lib,
+  ninja,
+  symlinkJoin,
+  which,
+  pythonPackages,
+  pythonSupport ? false,
+}:
+assert lib.asserts.assertMsg
+(pythonPackages.torch.cudaSupport)
+"tiny-cuda-nn requires torch to be built with cudaSupport"; let
+  inherit (lib) lists strings;
+  inherit (pythonPackages.torch) cudaPackages cudaSupport;
+  inherit (cudaPackages) cudatoolkit cudaFlags;
+
+  cuda-redist = symlinkJoin {
+    name = "cuda-redist";
+    paths = with cudaPackages; [
+      cuda_cudart
+      cuda_nvcc
+      libcublas
+      libcusparse
+      libcusolver
+    ];
+  };
+in
+  stdenv.mkDerivation (finalAttrs: {
+    name = "tiny-cuda-nn";
+    version = "1.6";
+
+    format = strings.optionalString pythonSupport "setuptools";
+
+    src = fetchFromGitHub {
+      owner = "NVlabs";
+      repo = finalAttrs.name;
+      rev = "v${finalAttrs.version}";
+      fetchSubmodules = true;
+      hash = "sha256-qW6Fk2GB71fvZSsfu+mykabSxEKvaikZ/pQQZUycOy0=";
+    };
+
+    nativeBuildInputs =
+      [
+        which
+        cmake
+        ninja
+      ]
+      ++ lists.optionals pythonSupport (with pythonPackages; [
+        pip
+        wheel
+        setuptools
+      ]);
+
+    buildInputs =
+      [
+        cuda-redist
+      ]
+      ++ lib.optionals pythonSupport (
+        with pythonPackages; [
+          python
+          pybind11
+        ]
+      );
+
+    # TODO: OpenCV, whicih also provides python bindings and supports CUDA, adds the cuda
+    #   redistributable to propagatedBuildInputs. Should we do this here as well?
```

Comment:
short answer: no, hopefully not. if you need cuda stuff at runtime the best solution is to [symlink stuff into $out as is done in the jaxlib-bin](https://github.com/NixOS/nixpkgs/blob/28c1aac72e3aef70b8c898ea9c16d5907f9eae22/pkgs/development/python-modules/jaxlib/bin.nix#L112-L118)